### DISCUSSION TITLE
New DatePicker Component

### DISCUSSION THREAD

**Discussion by ajg102** [tailwindlabs/headlessui#1979] [Ideas]
Hello,

I was recently looking into libraries for DatePicker components and realized there isn't any solution that is truly ideal for most projects. The three main solutions right now are to use a UI library, however they are usually opinionated in their styles and might not fit with your project's theme. You can use the browser default and style the input, but you lose control over the style of the default browser date/time dialog. The last option is to go custom, but that can be tedious and time consuming to implement all of the date specific logic.

It would be awesome for headlessui to include a DatePicker component that handled all of the date work (can use helpers from date-fns) in a headless fashion so developers only needed to worry about implementation and styling.

I'd be happy to help contribute toward its development. It should fundamentally work pretty similarly to the comboxbox, just with extra actions and render props available.

Overall, love working with headlessui and would be great to get a component like this included as well.

**Comment by nicolas-bastien** [comment]
Hi, I agree with @ajg102.

Since we are working with tailwindui setup, we are more confident in using headlessui component.
Datepicker would be a great one to add.
